src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,476 @@
+  }
+
+  createAndSetDataReader(divContainer, height, margin) {
+    const vertical = divContainer
+      .insert("div", "#svgContainer")
+      .attr("id", "vertical")
+      .style("position", "absolute") // need relative in parent
+      .style("z-index", "19")
+      .style("width", "1px")
+      .style("height", `${height}px`)
+      .style("top", "10px")
+      .style("bottom", "30px")
+      .style("left", "0px")
+      .style("margin-top", `${margin.top - 10}px`)
+      .style("pointer-events", "none")
+      .style("background", "#000");
+
+    const tooltip = divContainer
+      .insert("div", "#svgContainer")
+      .attr("class", `remove ${css(styles.tooltip)}`)
+      .attr("id", "tooltip")
+      .style("position", "absolute")
+      .style("z-index", "2")
+      .style("visibility", "hidden")
+      .style("top", "4px")
+      .style("left", "12px");
+
+    return [tooltip, vertical];
+  }
+
+  createAndSetTheLegend(d3DivContainer, colorNameList) {
+    const legendView = d3DivContainer
+      .append("div")
+      .attr("id", "legend")
+      .attr("class", `${css(styles.legend)}`);
+
+    for (let index = colorNameList.length - 1; index > 0; index--) {
+      let element = colorNameList[index];
+      let legendItemContainer = legendView
+        .append("div")
+        .attr("class", `${css(styles.legendItem)}`);
+
+      legendItemContainer
+        .append("svg")
+        .attr("width", 15)
+        .attr("height", 15)
+        .style("background-color", element[1]);
+
+      legendItemContainer
+        .append("text")
+        .attr("class", `${styles.legendItemText}`)
+        .text(element[0]);
+    }
+  }
+
+  setViewportEvents(svgViewport, vertical) {
+    let componentContext = this;
+    svgViewport
+      .on("mouseover", function(d, i) {
+        if (componentContext.mouseIsOverStream) {
+          let mousex = d3.mouse(this);
+          mousex = mousex[0];
+          vertical.style("visibility", "visible");
+          vertical.style("left", mousex + "px");
+        } else {
+          vertical.style("visibility", "hidden");
+        }
+      })
+      .on("mousemove", function(d, i) {
+        if (componentContext.mouseIsOverStream) {
+          let mousex = d3.mouse(this);
+          mousex = mousex[0];
+          vertical.style("visibility", "visible");
+          vertical.style("left", mousex + "px");
+        } else {
+          vertical.style("visibility", "hidden");
+        }
+      });
+  }
+
+  setMouseMoveAndOverStreams(tooltip) {
+    let componentContext = this;
+
+    this.streams
+      .on("mousemove", function(d, i) {
+        let mousex = d3.mouse(this)[0];
+        let date = componentContext.xAxisScale.invert(mousex);
+        componentContext.mouseIsOverStream = true;
+        this.hoveredKey = d3.select(this).attr("name");
+
+        for (let elem of d) {
+          if (elem.data.date.getFullYear() == date.getFullYear()) {
+            this.hoveredValue = elem.data[this.hoveredKey];
```

Comment:
For loops are usually not readable.
You can replace this whole loop by one line:

```js
const hoveredValue = d
    .find(elem => elem.data.date.getFullYear() === date.getFullYear())
    .data[hoveredKey];
```
PS: In JS `==` is not safe, please use `===`